library/src/main/java/com/datatorrent/lib/io/block/ReaderContext.java

```diff
@@ -190,11 +190,11 @@ protected Entity readEntity() throws IOException
           if (bytesRead == -1) {
             break;
           }
-          strBuffer = new String(buffer);
+          bufferStr = new String(buffer,0, bytesRead);
         }
 
-        while (posInStr < strBuffer.length()) {
-          char c = strBuffer.charAt(posInStr);
+        while (posInStr < bufferStr.length()) {
+          char c = bufferStr.charAt(posInStr);
           if (c != '\r' && c != '\n') {
             tmpBuilder.write(c);
```

Comment:
@tweise I will prefer to work on it later. Also, I have some possible alternative in my mind which I will propose in the new JIRA. 
I have created new JIRA APEXMALHAR-2198 for this.